### PR TITLE
Add scoped_set_flag cpp counterpart which already exists in Python

### PR DESCRIPTION
**Pull request by rtabbara**
In Python we could do something like

```
with dr.scoped_set_flag(dr.JitFlag.LoopRecord, False):
 ...
```
Adding an equivalent in cpp

```
{
   dr::scoped_set_flag scope(JitFlag::LoopRecord, false);
   ...
}
```

### CODE REVIEW

File: include/drjit/array_router.h [hunk 1]
```diff
@@ -1327,6 +1327,27 @@ DRJIT_INNER_REDUCTION(mean)
 
 #undef DRJIT_INNER_REDUCTION
 
+
```
Comment (njroussel) [inline include/drjit/array_router.h:1330]: Let's move it between `eval()` and `set_device` (line 1403 in this commit). You can turn your comment into a `\brief` documentation for the `struct`.

**Review by njroussel — APPROVED** [review]
Perfect! One tiny suggestion, feel free to merge the PR once you're done.

File: include/drjit/array_router.h [hunk 2]
```diff
@@ -1327,6 +1327,27 @@ DRJIT_INNER_REDUCTION(mean)
...
+// -------------------------------------------------------------------
+//! @{ \name Context manager for setting JIT flags
+// -------------------------------------------------------------------
+
+struct scoped_set_flag {
+    scoped_set_flag(JitFlag flag, bool value) : 
+        flag(flag),
+        orig_value(jit_flag(flag)) {
+
```
Comment (Speierers) [inline include/drjit/array_router.h:1339]: Remove empty line

File: include/drjit/array_router.h [hunk 3]
```diff
@@ -1327,6 +1327,27 @@ DRJIT_INNER_REDUCTION(mean)
...
+        jit_set_flag(flag, value);
+    }
+
+    ~scoped_set_flag() {
+        jit_set_flag(flag, orig_value);
+    }
+
+    JitFlag flag;
+    bool orig_value;
```
Comment (Speierers) [inline include/drjit/array_router.h:1348]: Minor:
```suggestion
    bool original_value;
```

**Review by Speierers — COMMENTED** [review]
LGTM! Just two minor comments

File: include/drjit/array_router.h (hunk 3 above)
Comment (rtabbara) [reply include/drjit/array_router.h]: Done!

**Pull request merged**